include/crocoddyl/core/diff-action-base.hxx

```diff
@@ -89,11 +89,189 @@ DifferentialActionModelAbstractTpl<Scalar>::createData() {
+  assert_pretty(is_a_AssignmentOp(op), ("op must be one of the AssignmentOp {settop, addto, rmfrom}"));
+  if (static_cast<std::size_t>(A.cols()) != state_->get_nv()) {
+    throw_pretty("Invalid argument: "
+                 << "number of columns of A is wrong, it should be " + std::to_string(state_->get_nv()) +
+                        " instead of " + std::to_string(A.cols()));
+  }
+  if (A.rows() != out.rows()) {
+    throw_pretty("Invalid argument: "
+                 << "A and out have different number of rows: " + std::to_string(A.rows()) + " and " +
+                        std::to_string(out.rows()));
+  }
+  if (static_cast<std::size_t>(out.cols()) != nu_) {
+    throw_pretty("Invalid argument: "
+                 << "number of columns of out is wrong, it should be " + std::to_string(nu_) + " instead of " +
+                        std::to_string(out.cols()));
+  }
+  switch (op) {
+    case setto:
+      out.noalias() = A * Fu;
+      break;
+    case addto:
+      out.noalias() += A * Fu;
+      break;
+    case rmfrom:
+      out.noalias() -= A * Fu;
+      break;
+    default:
+      throw_pretty("Invalid argument: allowed operators: setto, addto, rmfrom");
+  }
+}
+
+template <typename Scalar>
+void DifferentialActionModelAbstractTpl<Scalar>::multiplyFuTransposeBy(const Eigen::Ref<const MatrixXs>& FuTranspose,
+                                                                       const Eigen::Ref<const MatrixXs>& A,
+                                                                       Eigen::Ref<MatrixXsRowMajor> out,
+                                                                       const AssignmentOp op) const {
+  assert_pretty(is_a_AssignmentOp(op), ("op must be one of the AssignmentOp {settop, addto, rmfrom}"));
+  if (static_cast<std::size_t>(A.rows()) != state_->get_nv()) {
+    throw_pretty("Invalid argument: "
+                 << "number of rows of A is wrong, it should be " + std::to_string(state_->get_nv()) + " instead of " +
+                        std::to_string(A.rows()));
+  }
+  if (A.cols() != out.cols()) {
+    throw_pretty("Invalid argument: "
+                 << "A and out have different number of columns: " + std::to_string(A.cols()) + " and " +
+                        std::to_string(out.cols()));
+  }
+  if (static_cast<std::size_t>(out.rows()) != nu_) {
+    throw_pretty("Invalid argument: "
+                 << "number of rows of out is wrong, it should be " + std::to_string(nu_) + " instead of " +
+                        std::to_string(out.cols()));
+  }
+  switch (op) {
+    case setto:
+      out.noalias() = FuTranspose * A;
+      break;
+    case addto:
+      out.noalias() += FuTranspose * A;
+      break;
+    case rmfrom:
+      out.noalias() -= FuTranspose * A;
+      break;
+    default:
+      throw_pretty("Invalid argument: allowed operators: setto, addto, rmfrom");
+  }
```

Comment:
Why this set of lines of code is duplicated everywhere?

Reply:
There are two kind of operators:
  1. A * Jacobian
  2. Jacobian transpose * A

where the Jacobians could be `Fx` or `Fu`.
The latter uses a row-major storage, instead the former uses a column-major one.
As far I understand, we can try to condense both functions into a single one, but then, we'll affect the performance.
What are your thoughts/feedback about it?
(Note that the control-parametrization class follows the same pattern)

Additionally, these operators are defined in two classes:
  1. action model
  2. diff-action model

This is needed as we want to exploit the structure of both: integrators and differential action models.

Finally, this set of lines (as the one in the action-model class) describe a default implementation, which it is a `dense-multiplication` operator.
In short, we also need them.

Reply:
@jcarpent  -- is there a follow-up point to discuss?